src/extension/kernel.ts

```diff
@@ -1102,4 +1102,81 @@ export class Kernel implements Disposable {
   public getPlainCache(cacheId: string): Promise<Uint8Array> | undefined {
     return this.serializer?.getPlainCache(cacheId)
   }
+
+  runProgram(program?: RunProgramOptions | string) {
+    return new Promise<{ exitReason?: RunnerExitReason; code?: number | void; output?: string }>(
```

Comment:
@pastuxso this is still not right. i'm going to send a PR fixing `resouce.ts` that you can model it after.

Reply:
https://github.com/stateful/vscode-runme/pull/1582